programs/github

```diff
@@ -21,13 +21,48 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local acceptedFlags = {
+	'-b' = 'branch',
+	'--branch' = 'branch',
+	'-t' = 'tag',
+	'--tag' = 'tag',
+	'-a' = 'auth',
+	'--auth' = 'auth'
+}
+
+local findFlag = function(arg) return acceptedFlags[arg] end
+
+local passed = {...}
+local args = {}
+local flags = {}
+local flagValue = ''
+for k, arg in pairs(passed) do
+	local flagName = findFlag(arg)
+	if flagName and arg ~= flagValue then
+		flags[flagName] = passed[k + 1]
+		flagValue = flags[flagName]
+	elseif arg ~= action and arg ~= flagValue then
+		table.insert(args, arg)
+	end
+end
+
+if flags.branch and flags.tag then
+	return printError('please specify a branch or tag.')
+end
+
 if action == 'clone' then
 	-- parse input
-	local repo, treeName, dest = select(2, ...)
+	local repo, dest = args[1], args[2]
+	local treeName = flags.branch or flags.tag or 'master'
 
-	if not dest then
-		dest = treeName
-		treeName = 'master'
+	if flags.auth then
+		local authUser, authPass = flags.auth:match('([^:]+):([^:]+)')
+		if not authUser or not authPass then
+			return printError('Invalid username password - should be username:password')
+		else
+			github.basicAuth(authUser,authPass)
```

Comment:
Or rather, a better way to phrase the question. Each reference to `github` is calling the same table, correct?